saleor/graphql/page/dataloaders.py

```diff
@@ -108,11 +114,13 @@ class AssignedPageAttributesByPageIdLoader(DataLoader):
     def batch_load(self, keys):
         requestor = get_user_or_app_from_context(self.context)
         if requestor.is_active and requestor.has_perm(PagePermissions.MANAGE_PAGES):
-            qs = AssignedPageAttribute.objects.all()
+            qs = AssignedPageAttribute.objects.using(
+                self.database_connection_name
+            ).all()
         else:
-            qs = AssignedPageAttribute.objects.filter(
-                assignment__attribute__visible_in_storefront=True
-            )
+            qs = AssignedPageAttribute.objects.using(
+                self.database_connection_name
+            ).filter(assignment__attribute__visible_in_storefront=True)
```

Comment:
maybe it's worth to create a separate ticket to have a closer look at this (performance wise)? that's a join on a join and we're using that 3 times across this PR, might be beneficial to subquery it.

Reply:
We have a task for optimizing the Page Filters.